text/0000-io-fs-2.1.md

```diff
@@ -0,0 +1,576 @@
+of time stamp or moment in time, both of which are difficult abstractions to add
+lightly.
+
+Consequently, both of these functions will be **deprecated** in favor of
+requiring platform-specific code to access the modification/access time of
+files. This information is all available via the `MetadataExt` traits listed
+above.
+
+Eventually, once a `std` type for cross-platform timestamps is available, these
+methods will be re-instated as returning that type.
+
+### Lowering and setting `Permissions` (Unix)
+
+> **Note**: this section only describes behavior on unix.
+
+Currently there is no stable method of inspecting the permission bits on a file,
+and it is unclear whether the current unstable methods of doing so,
+`PermissionsExt::mode`, should be stabilized. The main question around this
+piece of functionality is whether to provide a higher level abstractiong (e.g.
+similar to the `bitflags` crate) for the permission bits on unix.
+
+This RFC proposes considering the methods for stabilization as-is and not
+pursuing a higher level abstraction of the unix permission bits. To facilitate
+in their inspection and manipulation, however, the following constants will be
+added:
+
+```rust
+mod os::unix::fs {
+    pub const USER_READ: raw::mode_t;
+    pub const USER_WRITE: raw::mode_t;
+    pub const USER_EXECUTE: raw::mode_t;
+    pub const USER_RWX: raw::mode_t;
+    pub const OTHER_READ: raw::mode_t;
+    pub const OTHER_WRITE: raw::mode_t;
+    pub const OTHER_EXECUTE: raw::mode_t;
+    pub const OTHER_RWX: raw::mode_t;
+    pub const GROUP_READ: raw::mode_t;
+    pub const GROUP_WRITE: raw::mode_t;
+    pub const GROUP_EXECUTE: raw::mode_t;
+    pub const GROUP_RWX: raw::mode_t;
+    pub const ALL_READ: raw::mode_t;
+    pub const ALL_WRITE: raw::mode_t;
+    pub const ALL_EXECUTE: raw::mode_t;
+    pub const ALL_RWX: raw::mode_t;
+    pub const SETUID: raw::mode_t;
+    pub const SETGID: raw::mode_t;
+    pub const STICKY_BIT: raw::mode_t;
+}
+```
+
+Finally, the `set_permissions` function of the `std::fs` module is also proposed
+to be marked `#[stable]` soon as a method of blanket setting permissions for a
+file.
+
+## Constructing `Permissions`
+
+Currently there is no method to construct an instance of `Permissions` on any
+platform. This RFC proposes adding the following APIs:
+
+```rust
+mod os::unix::fs {
+    pub trait PermissionsExt {
+        fn from_mode(mode: raw::mode_t) -> Self;
+    }
+    impl PermissionsExt for Permissions { ... }
+}
+```
+
+This RFC does not propose yet adding a cross-platform way to construct a
+`Permissions` structure due to the radical differences between how unix and
+windows handle permissions.
+
+## Creating directories with permissions
+
+Currently the standard library does not expose an API which allows setting the
+permission bits on unix or security attributes on Windows. This RFC proposes
+adding the following API to `std::fs`:
+
+```rust
+pub struct CreateDirOptions { ... }
```

Comment:
Yeah I do find `CreateDirOptions` to be a bit wordy, `DirOptions` or `DirBuilder` does sound good to me. I think though that naming-wise this may want to defer to https://github.com/rust-lang/rfcs/pull/1036 as the outcome of that RFC will likely clearly decide this builder's name.